assignments/prototype-refactor.js

```diff
@@ -7,3 +7,151 @@ Prototype Refactor
 2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
 
 */
+
+/*
+  Object oriented design is commonly used in video games.  For this part of the assignment you will be implementing several constructor functions with their correct inheritance hierarchy.
+  In this file you will be creating three constructor functions: GameObject, CharacterStats, Humanoid.  
+  At the bottom of this file are 3 objects that all end up inheriting from Humanoid.  Use the objects at the bottom of the page to test your constructor functions.
+  
+  Each constructor function has unique properties and methods that are defined in their block comments below:
+*/
+  
+/*
+  === GameObject ===
+  * createdAt
+  * dimensions (These represent the character's size in the video game)
+  * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
+*/
+
+class GameObject {
+    constructor (attributes){
+        this.createdAt = attributes.createdAt;
+        this.dimensions =  attributes.dimensions;
+    }
+
+  destroy(){
+      return 'Object was removed from the game.'
+  }
+  
+}
+  /*
+    === CharacterStats ===
+    * healthPoints
+    * name
+    * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+    * should inherit destroy() from GameObject's prototype
+  */
+  
+    class CharacterStats extends GameObject {
+        constructor(charAttributes){
+            super(charAttributes);
+            this.healthPoints = charAttributes.healthPoints;
+            this.name = charAttributes.name;
+        }
+
+        takeDamage(){
+            return `${this.name} took damage.`;
+        }
+    }
+  
+  /*
+    === Humanoid (Having an appearance or character resembling that of a human.) ===
+    * team
+    * weapons
+    * language
+    * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+    * should inherit destroy() from GameObject through CharacterStats
+    * should inherit takeDamage() from CharacterStats
+  */
+  
+  class Humanoid extends CharacterStats {
```

Comment:
Nice class constructor object, use of `this` keyword and string template literals